packages/terra-functional-testing/src/reporters/file-output-reporter/index.js

```diff
@@ -0,0 +1,205 @@
+    this.setTestModule = this.setTestModule.bind(this);
+    this.printReport = this.printReport.bind(this);
+    this.getMessage = this.getMessage.bind(this);
+
+    this.setResultsDir();
+    this.hasResultsDir();
+  }
+
+  /**
+   * Sets results directory for the test run. Uses the wdio reporterOptions.outputDir if set, otherwise
+   * it outputs to tests/wdio/reports.
+   * @return null;
+   */
+  setResultsDir() {
+    this.resultsDir = path.join(process.cwd(), 'tests', 'wdio', 'reports');
+  }
+
+  /**
+   * Check and create reports dir if doesn't exist
+   * @return null
+   */
+  hasResultsDir() {
+    if (!fs.existsSync(this.resultsDir)) {
+      fs.mkdirSync(this.resultsDir, { recursive: true }, (err) => {
+        if (err) {
+          Logger.error(err.message, { context: LOG_CONTEXT });
+        }
+      });
+    }
+  }
+
+  /**
+   * Formatting the filename based on locale, theme, and formFactor
+   * @return null
+   */
+  fileNameCheck({ formFactor, locale, theme }, browserName, cid) {
+    const fileNameConf = ['fileOutput'];
+    if (locale) {
+      fileNameConf.push(locale);
+      this.resultJsonObject.locale = locale;
+    }
+
+    if (theme) {
+      fileNameConf.push(theme);
+      this.resultJsonObject.theme = theme;
+    }
+
+    if (formFactor) {
+      fileNameConf.push(formFactor);
+      this.resultJsonObject.formFactor = formFactor;
+    }
+
+    if (browserName) {
+      fileNameConf.push(browserName);
+    }
+
+    if (cid) {
+      fileNameConf.push(cid);
+    }
+
+    this.fileName = fileNameConf.join('-');
+  }
+
+  /**
+   * Set the package name to moduleName property if specsValue contains /package string
+   * @param {string} specsValue - File path of current spec file from runners
+   * @return null
+   */
+  // eslint-disable-next-line class-methods-use-this
+  setTestModule(spec) {
+    if (spec.indexOf(`${path.sep}packages${path.sep}`) > -1) {
+      return spec.split(`${path.sep}packages${path.sep}`).pop().split(path.sep)[0];
+    }
+    const packageJsonPath = path.join(process.cwd(), 'package.json');
+    if (fs.existsSync(packageJsonPath)) {
+      const packagejson = fs.readJsonSync(packageJsonPath);
+      return packagejson.name;
+    }
+    return process.cwd().split(path.sep).pop();
+  }
+
+  onRunnerEnd(runner) {
+    this.runners.push(runner);
+    this.printReport(runner);
+  }
+
+  getMessage(runner) {
+    const results = this.getResultDisplay();
+    if (results.length === 0) {
+      return null;
+    }
+    const testLinks = runner.isMultiremote
+      ? Object.entries(runner.capabilities).map(
+        ([instanceName, capabilities]) => this.getTestLink({
+          config: { ...runner.config, ...{ capabilities } },
+          sessionId: capabilities.sessionId,
+          isMultiremote: runner.isMultiremote,
+          instanceName,
+        }),
+      )
+      : this.getTestLink(runner);
+    const output = [
+      ...this.getHeaderDisplay(runner),
+      '',
```

Comment:
my vote is to convert this multi line ternary into a regular if else statement, it's kind of hard to follow :/

Reply:
Updated: https://github.com/cerner/terra-toolkit/commit/979bbe09908240c56f67a200b4457942fa47dda6